packages/consumption/src/modules/requests/itemProcessors/proposeAttribute/ProposeAttributeRequestItemProcessor.ts

```diff
@@ -93,15 +111,60 @@ export class ProposeAttributeRequestItemProcessor extends GenericRequestItemProc
         _requestItem: ProposeAttributeRequestItem,
         params: AcceptProposeAttributeRequestItemParametersJSON,
         requestInfo: LocalRequestInfo
-    ): Promise<ProposeAttributeAcceptResponseItem> {
-        const parsedParams: AcceptProposeAttributeRequestItemParameters = AcceptProposeAttributeRequestItemParameters.from(params);
+    ): Promise<ProposeAttributeAcceptResponseItem | AttributeSuccessionAcceptResponseItem> {
+        const parsedParams = AcceptProposeAttributeRequestItemParameters.from(params);
 
         let sharedLocalAttribute: LocalAttribute;
         if (parsedParams.isWithExistingAttribute()) {
-            sharedLocalAttribute = await this.copyExistingAttribute(parsedParams.attributeId, requestInfo);
-        } else {
-            sharedLocalAttribute = await this.createNewAttribute(parsedParams.attribute!, requestInfo);
+            const existingSourceAttribute = await this.consumptionController.attributes.getLocalAttribute(parsedParams.attributeId);
+            if (typeof existingSourceAttribute === "undefined") {
+                throw TransportCoreErrors.general.recordNotFound(LocalAttribute, parsedParams.attributeId.toString());
+            }
+
+            const latestSharedVersion = await this.consumptionController.attributes.getSharedVersionsOfRepositoryAttribute(parsedParams.attributeId, [requestInfo.peer], true);
+
+            if (latestSharedVersion.length === 0) {
+                sharedLocalAttribute = await this.copyExistingAttribute(existingSourceAttribute.id, requestInfo);
```

Comment:
What's the point of `copyExistingAttribute`? It wraps a single call to `createSharedLocalAttributeCopy` and is called only once. Might as well yeet it and insert its implementation here.

Reply:
Using helper functions with descriptive names to avoid evil comments is prefectly fine. Just in this particular instance it's a downgrade compared to a simple call to `createSharedLocalAttributeCopy` since all you did is to burden yourself with the task of finding another descriptive name for a function that already exists. _Create a shared attribute copy_ is a more accurate description of what's going on than _copy existing attribute_, which suggested to me that we're duplicating the attribute for some reason.

My 2 cents, nothing dramatic. Keep it, if you feel strongly about it.